File: campuslife/ProfileViewController.m
//
//  ProfileViewController.m
//  LCSC
//
//  Created by Eric de Baere Grassl on 2/29/16.
//  Copyright © 2016 LCSC. All rights reserved.
//

#import "ProfileViewController.h"
#import "LCSC-Swift.h"


@interface ProfileViewController ()
@property (weak, nonatomic) IBOutlet UITextField *usernameDisplay;
@property (weak, nonatomic) IBOutlet UITextField *passwordDisplay;
@property (strong, nonatomic) Authentication *auth;
@property (strong, nonatomic) UIAlertView *alert;
@end

@implementation ProfileViewController


- (void)changeDisplaytext:(NSString*)newLogin andPassword:(NSString*)newPassword{
    _usernameDisplay.text = newLogin;
    _passwordDisplay.text = newPassword;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    _auth = [[Authentication alloc] init];
    NSString *username = [_auth getUsername];
    NSString *password = [_auth getPassword];
    [self changeDisplaytext:username andPassword:password];
}

- (IBAction)clearTapped:(id)sender {
    [_auth clearProfile];
    [self changeDisplaytext:@"" andPassword:@""];
}

- (IBAction)signingTapped:(id)sender {
    if ([_auth setProfile:_usernameDisplay.text newPassword:_passwordDisplay.text]){
        NSString *newLogin = [_auth getUsername];
        NSString *newPassword = [_auth getPassword];
        [self changeDisplaytext:newLogin andPassword:newPassword];
        _alert = [[UIAlertView alloc] init];
        _alert.title = @"Success!";
        _alert.message = @"Your username and password were saved.";
        _alert.delegate = self;
        [_alert addButtonWithTitle:@"Ok"];
        [_alert show];
        [self performSegueWithIdentifier:@"signinTapped" sender:sender];
    } else {
        _alert = [[UIAlertView alloc] init];
        _alert.title = @"Fail!";
        _alert.message = @"Your username and password have or invalide input.";
        _alert.delegate = self;
        [_alert addButtonWithTitle:@"Ok"];
        [_alert show];
    }
    
}

- (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    if (![_auth setProfile:_usernameDisplay.text newPassword:_passwordDisplay.text]){
        NSString *newLogin = [_auth getUsername];
        NSString *newPassword = [_auth getPassword];
        [self changeDisplaytext:newLogin andPassword:newPassword];
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

@end

File: campuslife/ProfileViewController.swift
//
//  ProfileViewController.swift
//  LCSC
//
//  Created by Eric de Baere Grassl on 3/23/16.
//  Copyright © 2016 LCSC. All rights reserved.
//

import UIKit

class ProfileViewController: UIViewController,UINavigationControllerDelegate, UIImagePickerControllerDelegate, ImageCropViewControllerDelegate {
    @IBOutlet weak var menuButton: UIBarButtonItem!
    
    @IBOutlet weak var imageView: UIImageView!
    var imagePicker: UIImagePickerController!
    let auth = Authentication()
    
    
    
    @IBAction func cameraButtonTapped(sender: UIBarButtonItem) {
        imagePicker = UIImagePickerController()
        imagePicker.delegate = self
        imagePicker.sourceType = .Camera
        presentViewController((imagePicker), animated: true, completion: nil)
    }
    
    func promptAlet(title: String, message: String){
        let alert:UIAlertView = UIAlertView()
        alert.title = title
        alert.message = message
        alert.delegate = self
        alert.addButtonWithTitle("Ok")
        alert.show()
    }
    
    //save the card image
    func saveImage(image: UIImage){
        NSUserDefaults.standardUserDefaults().setObject(UIImageJPEGRepresentation(image,1), forKey: "card")
        NSUserDefaults.standardUserDefaults().synchronize()
    }
    
    //allows the user to crop the image
    func ImageCropViewControllerSuccess(controller: UIViewController!, didFinishCroppingImage croppedImage: UIImage!) {
        saveImage(croppedImage)
        promptAlet("Success!", message: "Your card picture was saved.")
        checkAndLoadCardPicture()
        navigationController?.popViewControllerAnimated(true)
    }
    
    
    func ImageCropViewControllerDidCancel(controller: UIViewController!) {
        navigationController?.popViewControllerAnimated(true)
    }
    
    //saves the photo taken by the user
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
        imagePicker.dismissViewControllerAnimated(true, completion: nil)
        let image = (info[UIImagePickerControllerOriginalImage] as? UIImage)!
        saveImage(image)
        checkAndLoadCardPicture()
        let controller = ImageCropViewController(image: image)
        controller.delegate = self
        navigationController?.pushViewController(controller, animated: true)
    }
    
    //load the card image in case it exists
    func checkAndLoadCardPicture(){
        imageView.contentMode = UIViewContentMode.ScaleAspectFit
        if let imageToLoad = NSUserDefaults.standardUserDefaults().objectForKey("card"){
            imageView.image = UIImage(data: imageToLoad as! NSData)
        } else {
            imageView.image = UIImage(named: "squirrelCard")
            if !auth.userHaveEverBeenAtProfilePage(){
                promptAlet("No card picture is registered!", message: "You can register your card picture by tapping the camera button.")
                auth.setUserHaveEverBeenAtProfilePage(true)
            }
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        //my code :)
        menuButton.target = revealViewController()
        menuButton.action = Selector("revealToggle:")
        
        imageView.layer.cornerRadius = 5
        imageView.clipsToBounds = true
        imageView.layer.borderColor = UIColor.blackColor().CGColor
        imageView.layer.borderWidth = 2.0
        self.view.backgroundColor = UIColor(patternImage: UIImage(named: "backgroundCollor")!)
        self.view.addGestureRecognizer(self.revealViewController().panGestureRecognizer())
        
        self.view.addGestureRecognizer(self.revealViewController().tapGestureRecognizer())
        checkAndLoadCardPicture()
    }
}
